### ISSUE TITLE
*.3dja format for 'print3d'?

### ISSUE THREAD

**Issue by SammyForReal** [SwitchCraftCC/sc-peripherals#33]
So originally, there only were the `.2dj` and `.3dj` format for 3d and 2d prints. The community later made an `.2dja` format for bigger posters and it actually got implemented in the `print2d software too.` _(See #23)_

I now was wondering if a `.3dja` format would make sense to support in the `print3d` program too.
At this point, we could probably also try to figure out how to combine both programs _(print2d and print3d)_ into one as there wouldn't be any big differences in both implementations. _(Maybe within aliases for backwards compatibility?)_

It also got mentioned once that making both programs acting the same way _(at least in what they are showing on terminal)_ is on todo too, so this would be the perfect moment to do so actually.

I am opening this issue to get feedback about this and if it is something someone _(like me)_ could make an PR for.

**Comment by Wojbie** [comment]
I am sorry `.3dja` ? I was not aware community even standardized `3dja` format yet? Could you link example software that creates `.3dja` files?

**Comment by SammyForReal** [comment]
Maybe I was not clear enough sorry. No there is no .3dja file, this issue is about if we should introduce this standard however to make this process more similar to .2dja files and easier for multi-structure prints as well.

**Comment by ShreksHellraiser** [comment]
Support was added for 2dja because my solution to downloading multiple 2djs was to toss them in a single file as a 2dja. It ended up becoming a fairly popular way of managing multiple posters and support was PRd in. 

With that background, there's no existing software that has attempted a format like 3dja, so it currently doesn't exist.

**Comment by Magicmaan** [comment]
3d models work pretty differently. It isn't just a set of pixels to then randomly segment into different blocks. We'd need to make a lua program (or blockbench plugin perhaps) that is able to segment a larger model into the 16x16 chunks. But yeah, it'd be sick to have a 3dja format or something to be able to print multiblocks with ease

**Comment by SammyForReal** [comment]
Yeah the question is what are the specifications for such a format.

But I agree that this could get usefull. I would start implementing that to the sam3dj plugin on blockbench.

**Comment by bananasov** [comment]
> Yeah the question is what are the specifications for such a format.
> 
> But I agree that this could get usefull. I would start implementing that to the sam3dj plugin on blockbench.

You could just take a page out of `.2dja`s book and just have an array of objects inside a field just like `2dja`, of course with the needed metadata such as `title`, `tooltip` and etcetera.

**Comment by dimaguy** [comment]
I've written out a 3dja format definition for CCSMB (CCSMB/Standards#30), maybe it could be used for print3d if supported by the community

**Comment by scmcgowen** [comment]
I also wrote one earlier 
https://gist.github.com/scmcgowen/c245c527ad286281b7f5198bdce102ae

**Comment by dimaguy** [comment]
> I also wrote one earlier 
> https://gist.github.com/scmcgowen/c245c527ad286281b7f5198bdce102ae

Yours looks fine and probably somewhat compatible with mine(didnt look too thoroughly) but it's lacking the optimizations Ive thought of.
If you've done a proper implementation already I could give a hand with it
